### PR TITLE
TNT-43012 Return default content items

### PR DESCRIPTION
**Pull request by XDex**
Note: DO NOT MERGE until target-upstream v0.0.56 with support for `characteristics -> scopeType` is released to production.
## Description

Support default content proposition items delivered for control experiences with no offer content.

## Related Issue

https://jira.corp.adobe.com/browse/TNT-43012

## Motivation and Context

Support default content propositions in the response, so that customers know when a control experience is selected.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Personalization/createPersonalizationDetails.js
Comment (jfkhoury) [inline src/components/Personalization/createPersonalizationDetails.js:46]: I like the explicit nature of this PR.

I wish there was a way for Target Upstream to figure out which schemas to act on without a change to the Web SDK  which enable us to add more schemas in the future without a change to the SDKs..

**Review by jfkhoury — APPROVED** [review]

File: src/components/Personalization/createPersonalizationDetails.js
Comment (XDex) [reply src/components/Personalization/createPersonalizationDetails.js:46]: @jfkhoury We can make an optimization in this regard in case when content for all available schemas should be retrieved by not specifying `schemas` array at all in the request - in this case content for all available schemas will be returned implicitly from Target Upstream.
Thus, only when the page wide scope is not included in the request would we need to set the schemas explicitly to all values other than DOM_ACTION. Let me know if this would be desired and I'll open a ticket for it..

**Comment by 5amfung** [comment]
@XDex Your PR title is referring to the wrong TNT number.

**Review by ninaceban — CHANGES_REQUESTED** [review]
@XDex great job!
I saw you are checking that `renderAttempted=false` although there was nothing to render, and `renderAttempted` is confusing in this situation, it is an experience that was executed on the page because of the nature of `__view__` scope. The customers are chaining the `sendEvent` promise to filter the executed experiences, collect the response tokens and send them to third parties. 
I would say that in case `renderDecisions=true` we should add the `renderAttempted=true` flag and send a display notification.

**Comment by XDex** [comment]
@ninaceban As we discussed today, I'll wait for your A4T PR to get merged before adding the mods to mark default content items as renderable actions

File: test/functional/specs/Personalization/C28761.js
Comment (jonsnyder) [inline test/functional/specs/Personalization/C28761.js:80]: I'm wondering if renderAttempted should be true for default-content-items. In the documentation @ninaceban wrote for client side logging (https://git.corp.adobe.com/AdobeDocs/experience-platform.en/pull/2470), she recommends using the renderAttempted flag to determine which TNTAs/analyticsPayloads to record. If renderAttempted is false for default-content-items, client side logging would miss these.

File: src/components/Personalization/createPersonalizationDetails.js
Comment (jonsnyder) [reply src/components/Personalization/createPersonalizationDetails.js:46]: @XDex the optimization you suggested would still require a code change when a new schema was added. Target upstream should know that only when the "__view__" scope is requested, it should return DOM_ACTION schemas. If this were the case the Web SDK would only need to be updated when new schemas needed to be handled by the SDK. Currently only redirect, and dom action schemas are handled. The rest are passed through. 

If a new pass-through schema were added, this code would need to be updated even though the SDK isn't doing anything with it.

File: test/functional/specs/Personalization/C28761.js
Comment (jonsnyder) [reply test/functional/specs/Personalization/C28761.js:80]: Oh, I see Nina already brought up this concern.

File: src/components/Personalization/createPersonalizationDetails.js
Comment (XDex) [reply src/components/Personalization/createPersonalizationDetails.js:46]: Right, let's just leave it as-is then for now, the schemas being always explicitly specified in the request.

**Comment by XDex** [comment]
@ninaceban @jonsnyder I've implemented support for default-content items when `renderDecisions=true`, with proper grouping  to view/mbox scopes, please take another look when you get a chance.
Note - This PR requires the support of `characteristics -> scopeType` to be released in Target upstream before being merged, see https://jira.corp.adobe.com/browse/TNT-43696
Thank you!

File: test/functional/specs/Personalization/C28761.js
Comment (XDex) [reply test/functional/specs/Personalization/C28761.js]: Yes, this has been addressed in the latest changes

**Review by jonsnyder — APPROVED** [review]
These changes look good. I would like Nina to review before we merge though.

File: src/components/Personalization/groupDecisions.js
Comment (ninaceban) [inline src/components/Personalization/groupDecisions.js:125]: I would suggest to keep the old order of extracting the decisions:
1. Extract the `redirect` decisions by Redirect Schema(no changes needed).
2. Extract from remaining decisions from step 1 the `renderable` decisions (by DOM_ACTION, DEFAULT_CONTENT_ITEM schema and `scopeType: view`).
3. Extract `page-wide scope` decisions and `viewDecisions` from the `renderableDecisions` from step 2, by `__view__` scope.
4. Remaining decisions from step 2 are the form-based composed decisions.

Otherwise, it becomes extremely hard to understand why we group then concat at the end the decisions, then also maintain the code.

**Review by ninaceban — CHANGES_REQUESTED** [review]
@XDex Thank you Andrei. I see you have added a functional test,  could you also add unit tests that assert a proper grouping of the decisions based on the use-cases:
- default content item is returned for a view 
- default content item is returned for a decision scope
- default content item is returned for `__view__` scope

**Pull request merged**
